modules/stanford_courses/stanford_courses.install.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @file
+ * Stanford_courses.install.
+ */
+
+/**
+ * Drop the migrate_map_stanford_courses table and allow it to be recreated upon the next import.
+ */
+function stanford_courses_update_9001() {
+  $database = \Drupal::database();
+  if ($database->schema()->tableExists('migrate_map_stanford_courses')) {
+    // We need to remove previously imported courses and allow them to be recreated.
+    $query = $database->query("SELECT `destid1` FROM {migrate_map_stanford_courses}");
+    $result = $query->fetchAll();
+    foreach ($result as $obsolete_node_id) {
+      $node = \Drupal::entityTypeManager()->getStorage('node')->load($obsolete_node_id->destid1);
+      if ($node) {
+        $node->delete();
+      }
+    }
```

Comment:
Instead of this, you could just invoke the migration to rollback, but this will have large consequences for sites actively using the importer. There could be a scenario that the site managers tagged the courses with some taxonomy terms. It might be minimal and the number of sites using the importer is small enough we might be able to just communicate what's happening.
We should review with Sara/Ishi if that is acceptable.

Reply:
An alternative to deleting each node is a little more difficult but it can be done:

1. update the `migrate_map_stanford_courses` table to add the `sourceid2` and `sourceid3` columns
2. loop through each of the nodes, fetch the data from the API and set the subject and code values into the `sourceid2` and `sourceid3` columns respectively.

This has a caveat that if a course no longer exists in this years feed, it won't match.

Reply:
That second alternative also has the problem that stuff which was imported and which subsequently overwrote a previously imported course (which was a problem in the feed I've been testing with), then there will be mismatched data in one record, and possibly a duplication.

Reply:
I think the last time we checked, only about ten sites were using course importer thus far.  Any chance we can re-run that query?

Reply:
So I ran a set of queries across the production sites, and I found that only 17 sites are using courses (as of 10/17/2022), and none of them have any shared tags associated with the courses.

I think we're good to proceed with this approach.

I'm not sure why CodeClimate says 0% diff-coverage; I thought the procedural code (such as the `.install` file) were exempt from code coverage?